utils/s2n_mem.c

```diff
@@ -149,6 +149,10 @@ int s2n_mem_set_callbacks(s2n_mem_init_callback mem_init_callback, s2n_mem_clean
     return S2N_SUCCESS;
 }
 
+/*
+ * Generally, s2n_realloc is preferred over s2n_alloc. This is because calling
+ * s2n_alloc on a blob that already has memory allocated will leak memory.
+*/
```

Comment:
Nit: if you want people using the methods to see this comment, I'd put it in the header on the declaration, not here on the implementation.